File: app/src/main/java/com/mpandg/mpandgbluetooth/activity/SettingsActivity.java
package com.mpandg.mpandgbluetooth.activity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceActivity;
import android.preference.PreferenceManager;
import android.support.v7.app.AppCompatDelegate;
import android.view.MenuItem;

import com.mpandg.mpandgbluetooth.R;

public final class SettingsActivity extends PreferenceActivity
        implements SharedPreferences.OnSharedPreferenceChangeListener {

    private AppCompatDelegate mDelegate;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        //noinspection deprecation
        addPreferencesFromResource(R.xml.settings_activity);

        //noinspection ConstantConditions
        getDelegate().getSupportActionBar().setHomeButtonEnabled(true);
        getDelegate().getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        prefs.registerOnSharedPreferenceChangeListener(this);
        setPrefenceTitle(getString(R.string.pref_commands_mode));
        setPrefenceTitle(getString(R.string.pref_commands_ending));
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case android.R.id.home:
                finish();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }


    @Override
    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String value) {
        setPrefenceTitle(value);
    }


    /**
     * Set the list header.
     */
    private void setPrefenceTitle(String TAG) {
        //noinspection deprecation
        final Preference preference = findPreference(TAG);
        if (preference == null) return;
        if (preference instanceof ListPreference) {
            if (((ListPreference) preference).getEntry() == null) return;
            final String title = ((ListPreference) preference).getEntry().toString();
            preference.setTitle(title);
        }
    }

    private AppCompatDelegate getDelegate() {
        if (mDelegate == null) {
            mDelegate = AppCompatDelegate.create(this, null);
        }
        return mDelegate;
    }
}

File: app/src/main/java/com/mpandg/mpandgbluetooth/activity/SettingsActivity.kt
package com.mpandg.mpandgbluetooth.activity

import android.content.SharedPreferences
import android.content.SharedPreferences.OnSharedPreferenceChangeListener
import android.os.Bundle
import androidx.preference.ListPreference
import android.preference.PreferenceActivity
import androidx.preference.PreferenceManager
import android.view.MenuItem
import androidx.appcompat.app.AppCompatDelegate
import com.mpandg.mpandgbluetooth.R

class SettingsActivity : PreferenceActivity(), OnSharedPreferenceChangeListener {
    private var mDelegate: AppCompatDelegate? = null
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        addPreferencesFromResource(R.xml.settings_activity)
        delegate.supportActionBar!!.setHomeButtonEnabled(true)
        delegate.supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        prefs.registerOnSharedPreferenceChangeListener(this)
        setPreferenceTitle(getString(R.string.pref_commands_mode))
        setPreferenceTitle(getString(R.string.pref_commands_ending))
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.itemId
        return when (id) {
            android.R.id.home -> {
                finish()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences, value: String) {
        setPreferenceTitle(value)
    }

    /**
     * Set the list header.
     */
    private fun setPreferenceTitle(TAG: String) {
        val preference = findPreference(TAG) ?: return
        if (preference is ListPreference) {
            if (preference.entry == null) return
            val title = preference.entry.toString()
            preference.setTitle(title)
        }
    }

    private val delegate: AppCompatDelegate
        get() {
            if (mDelegate == null) {
                mDelegate = AppCompatDelegate.create(this, null)
            }
            return mDelegate!!
        }
}